project/scripts/genDocsScalaLang

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -e
+shopt -s extglob # needed for rm everything but x
+echo "Working directory: $PWD"
+
+GENDOC_EXTRA_ARGS=$@
+GIT_HEAD=$(git rev-parse HEAD) # save current head for commit message in gh-pages
+PREVIOUS_SNAPSHOTS_DIR="$PWD/../prev_snapshots"
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >& /dev/null && pwd)"
+SITE_OUT_DIR="$PWD/docs/_site"
+
+DOCS_SCALA_LANG_DIR="$PWD/docsScalaLang"
+
+rm -rf $DOCS_SCALA_LANG_DIR
+mkdir -pv $DOCS_SCALA_LANG_DIR
+git clone "https://github.com/scala/docs.scala-lang.git" $DOCS_SCALA_LANG_DIR
+
+SBT="$SCRIPT_DIR/sbt"
+mkdir -pv $SITE_OUT_DIR
+env "scaladoc.projectFormat=md" "$SBT" "scaladoc/renderScaladocScalajsToFile $DOCS_SCALA_LANG_DIR/scripts/scaladoc-scalajs.js $DOCS_SCALA_LANG_DIR/resources/css code-snippets.css"
+"dist/target/pack/bin/scaladoc" "-d" "$SITE_OUT_DIR" "-format" "md" "-siteroot" "docs" "/dev/null"
+
+if [ ! -d "$SITE_OUT_DIR" ]; then
+  echo "Output directory did not exist: $SITE_OUT_DIR" 1>&2
+  exit 1
+fi
+
+# Copy reference and scaladoc docs
+cp -rf "$SITE_OUT_DIR/docs/reference"/* "$DOCS_SCALA_LANG_DIR/_scala3-reference"
+cp -rf "$SITE_OUT_DIR/docs/usage/scaladoc"/* "$DOCS_SCALA_LANG_DIR/_overviews/scala3-scaladoc"
+
+cp -rf "$SITE_OUT_DIR/docs/reference/contextual/motivation.md" "$DOCS_SCALA_LANG_DIR/_scala3-reference/contextual.md"
+cp -rf "$SITE_OUT_DIR/docs/reference/metaprogramming/toc.md" "$DOCS_SCALA_LANG_DIR/_scala3-reference/metaprogramming.md"
+cp -rf "$SITE_OUT_DIR/docs/resources/talks.md" "$DOCS_SCALA_LANG_DIR/scala3/talks.md"
+cp -rf "$SITE_OUT_DIR/docs/usage/getting-started.md" "$DOCS_SCALA_LANG_DIR/scala3/getting-started.md"
+cp -rf "$SITE_OUT_DIR/docs/usage/language-versions.md" "$DOCS_SCALA_LANG_DIR/_scala3-reference/language-versions.md"
+cp -rf "$SITE_OUT_DIR/docs/usage/worksheet-mode.md" "$DOCS_SCALA_LANG_DIR/_overviews/scala3-book/tools-worksheets.md"
```

Comment:
I suggest that we host the files in the dotty repo with the same structure as in the docs.scala-lang.org repo (we should rename `docs/reference/contextual/motivation.md` to `docs/reference/contextual.md` here, for instance), and that we host here only the pages that are related to language features (everything in `reference/`), that contain code examples, or document a tool maintained in this repository, such as scaladoc. Thus, we won’t have to maintain all these `cp` commands (we would just `cp` the `reference` and `scaladoc`).

Reply:
You're right. I just made it that way to maintain compatibility, but in the future, we can reorganize the structure.